tuf/data/root.go

```diff
@@ -23,14 +24,53 @@ type Root struct {
 	ConsistentSnapshot bool                 `json:"consistent_snapshot"`
 }
 
+// isValidRoot returns an error, or nil, depending on whether the content of the struct
+// is valid for root metadata.  This does not check signatures or expiry, just that
+// the metadata content is valid.
+func isValidRoot(r Root) error {
```

Comment:
your comment is really helpful, but I'm wondering if we should also make it more explicit in the function name that we are only validating structure and not all content?  Maybe, `isValidRootStructure`?